src/platform_impl/macos/app_state.rs

```diff
@@ -180,13 +177,15 @@ impl Handler {
 pub enum AppState {}
 
 impl AppState {
-    pub fn set_callback<F, T>(callback: F, window_target: RootWindowTarget<T>)
+    pub unsafe fn set_callback<F, T>(callback: F, window_target: Rc<RootWindowTarget<T>>)
```

Comment:
Was following this [code](https://github.com/rust-windowing/winit/blob/31110be3960bb97caeacdd0a8cbad292722f94f1/src/platform_impl/windows/event_loop.rs#L334). However I do think that this function is unsafe, since it extends lifetime of callback, so users should be aware.

Reply:
Right, that makes sense. Adding a comment would be nice then.